### PR TITLE
SHIRO-712 - Add BasicIniEnvironment

### PR DESCRIPTION
**Pull request by coheigea**

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Shiro-pr/129/

File: core/src/main/java/org/apache/shiro/env/BasicIniEnvironment.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.apache.shiro.config.IniSecurityManagerFactory;
+import org.apache.shiro.mgt.SecurityManager;
+
+/**
+ * Basic usage:<p>
+ * <code>
+ * Environment env = new BasicIniEnvironment("classpath:shiro.ini");
+ * SecurityManager securityManager = env.getSecurityManager();
+ * </code>
+ *
```
Comment (bdemers) [inline core/src/main/java/org/apache/shiro/env/BasicIniEnvironment.java]: nit: add `@since 1.5`

**Comment by bdemers** [comment]
Nice!

File: core/src/main/java/org/apache/shiro/env/BasicIniEnvironment.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ */
+public class BasicIniEnvironment extends DefaultEnvironment {
+
+    public BasicIniEnvironment(Ini ini) {
+        setSecurityManager(new IniSecurityManagerFactory(ini).getInstance());
```
Comment (fpapon) [inline core/src/main/java/org/apache/shiro/env/BasicIniEnvironment.java]: `IniSecurityManagerFactory` is deprecated, @bdemers should we still use it?

File: core/src/main/java/org/apache/shiro/env/BasicIniEnvironment.java (hunk 1 above)
Comment (coheigea) [reply core/src/main/java/org/apache/shiro/env/BasicIniEnvironment.java]: @bdemers Done

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Shiro-pr/132/

File: core/src/main/java/org/apache/shiro/env/BasicIniEnvironment.java (hunk 2 above)
Comment (bdemers) [reply core/src/main/java/org/apache/shiro/env/BasicIniEnvironment.java]: Using it privately is fine, the `IniSecurityManagerFactory` code might not go away, probably just an implementation detail

Comment (fpapon) [reply core/src/main/java/org/apache/shiro/env/BasicIniEnvironment.java]: @bdemers ok :)

**Pull request merged**